### PR TITLE
Feature/refactor commands

### PR DESCRIPTION
**Pull request by stamp**

### CODE REVIEW

File: commands/reports/report_test.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package reports
+
+import (
+	"testing"
+
+	"github.com/stampzilla/gozwave/commands"
+	"github.com/stretchr/testify/assert"
+)
+
+type row struct {
```
Comment (jonaz) [inline commands/reports/report_test.go]: row seems to be used in only TestNew. Maybe put the declaration locally in there insead of out here?

File: commands/reports/switch_binary.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+type SwitchBinary struct {
+	*report
+	Status bool `json:"status"`
+	data   []byte
+}
+
+// NewSwitchBinary creates a new switchBinary report that contains the switch status from a zwave node
+func NewSwitchBinary(data []byte) (*SwitchBinary, error) {
+	if len(data) < 1 {
+		data = []byte{0x00}
```
Comment (jonaz) [inline commands/reports/switch_binary.go]: Not an error returned here? Is it not an error if len is < 1?

File: commands/reports/switch_multilevel.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+type SwitchMultilevel struct {
+	*report
+	Level byte `json:"level"`
+	data  []byte
+}
+
+func NewSwitchMultilevel(data []byte) (*SwitchMultilevel, error) {
+	if len(data) < 1 {
+		data = []byte{0x00}
```
Comment (jonaz) [inline commands/reports/switch_multilevel.go]: no error returned nere? Intentional?

File: commands/reports/report_test.go (hunk 1 above)
Comment (stamp) [inline commands/reports/report_test.go]: You are right :) moved it

File: commands/reports/switch_binary.go (hunk 2 above)
Comment (stamp) [inline commands/reports/switch_binary.go]: When I read the manual I discovered that we have totally missed version 2 of the switch_binary report. I added it, updated the tests and added error returns when length is invalid.

**Pull request merged**